### PR TITLE
PMM-10742 PMM-10470 PMM-10921 PMM-8553 PMM-10824 PMM-10944 Backups

### PR DESCRIPTION
**Pull request by artemgavrilov**
PMM-10742
PMM-10470
PMM-10921
PMM-8553
PMM-10824
PMM-10944

Build: SUBMODULES-2836

- [ ] Links to other linked pull requests (optional).

### CODE REVIEW

**Review by idoqo — APPROVED** [review]

File: managed/models/restore_history_helpers_test.go [hunk 1]
```diff
@@ -156,6 +161,8 @@ func TestRestoreHistory(t *testing.T) {
 		require.NoError(t, err)
 		assert.Equal(t, params.ArtifactID, i.ArtifactID)
 		assert.Equal(t, params.ServiceID, i.ServiceID)
+		// TODO looks like timezone of PITRTimestamp is changed after updating model in DB.
```
[the remark this answers is not in the document]
Comment (artemgavrilov) [reply managed/models/restore_history_helpers_test.go]: Here is the reson: https://github.com/stretchr/testify/issues/1010

Test fixed
cc @PavelKhripkov

**Comment by PavelKhripkov** [comment]
@artemgavrilov we still import `backupv1` as `backupv1beta1` in the code files like `pmm\agent\runner\jobs\mongodb_backup_job.go` and other. Let's fix that?

**Comment by artemgavrilov** [comment]
> @artemgavrilov we still import `backupv1` as `backupv1beta1` in the code files like `pmm\agent\runner\jobs\mongodb_backup_job.go` and other. Let's fix that?

Yes, fix it please

File: managed/services/backup/backup_service.go [hunk 2]
```diff
@@ -505,3 +526,78 @@ func (s *Service) prepareBackupJob(
...
+		return errors.Wrapf(ErrIncompatibleService, "restore to point in time is only available for MongoDB")
+	}
+
+	if artifact.Mode != models.PITR {
+		if pitrTimestamp.Unix() == 0 {
+			return nil
+		}
+		if pitrTimestamp.Unix() != 0 {
+			return errors.Wrapf(ErrIncompatibleArtifactMode, "artifact of type '%s' cannot be use to restore to point in time", artifact.Mode)
+		}
```
Comment (PavelKhripkov) [inline managed/services/backup/backup_service.go:581]:
```suggestion
		if pitrTimestamp.Unix() == 0 {
			return nil
		} else {
			return errors.Wrapf(ErrIncompatibleArtifactMode, "artifact of type '%s' cannot be use to restore to point in time", artifact.Mode)
		}
```

Comment (artemgavrilov) [reply managed/services/backup/backup_service.go]: Refactored

File: agent/runner/jobs/pbm_helpers.go [hunk 3]
```diff
@@ -180,128 +195,171 @@ func retrieveLogs(ctx context.Context, dbURL *url.URL, event string) ([]pbmLogEn
...
-			var status pbmStatus
-			if err := execPBMCommand(ctx, dbURL, &status, "status"); err != nil {
-				return errors.Wrapf(err, "pbm status error")
-			}
-			done, err := cond(status)
+			var info describeInfo
+			err := execPBMCommand(ctx, dbURL, &info, "describe-backup", name)
 			if err != nil {
-				return errors.Wrapf(err, "condition failed")
+				// for the first couple of seconds after backup process starts describe-backup command may return this error
```
[the remark this answers is not in the document]
Comment (artemgavrilov) [reply agent/runner/jobs/pbm_helpers.go:233]: We will address it later. After release we need to have retro with discussion about all tradeoffs that we have in code.

**Pull request merged**